sklego/feature_selection/mrmr.py

```diff
@@ -0,0 +1,219 @@
+    have high relevance to the target variable while minimizing redundancy among the already selected features.
+
+    How MRMR works:
+
+    1. Compute the relevance of each feature to the target variable: The relevance of a feature is typically
+    measured using a metric such as mutual information, correlation coefficient, or another appropriate measure of
+    dependence between the feature and the target variable.
+
+    2. Compute the redundancy between each pair of features: Redundancy is the degree of similarity or overlap between
+    features. It can be measured using metrics such as mutual information, correlation coefficient, or other similarity
+    measures.
+
+    3. Select features based on the maximum relevance and minimum redundancy criteria: MRMR aims to maximize the
+    relevance of selected features to the target variable while minimizing redundancy among them.
+
+    4. Construct the final subset of features: MRMR iteratively adds features to the selected subset until a predefined
+    number of features is reached.
+
+    The implemented formula is:
+    \text{score}_{i}(f) = \frac{\text{relevance}(f | y)}{\text{reduncancy}(f | \text{selected}_{i-1})}
+    Parameters
+    ----------
+    k : int
+        Number of feature the model should use.
+    relevance_func : str | Callable, default="f"
+        The relevance function to use. The default maps to scikit-learn [f_classif](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.f_classif.html) or  [f_regression](https://scikit-learn.org/stable/modules/generated/sklearn.feature_selection.f_regression.html) for classification or regression (resp.)
+    redundancy_func : str | Callable, default="p"
+        The redundancy function to use. The default maps to Pearson correlation computed for each remaining features.
+    kind : Literal["auto", "classficiation", "regression"], default="auto".
+        'classification' or 'regression' or 'auto' if auto the model
+        will try to infer the type of problem looking at the y data type, by default "auto".
+
+    !! warning:
+        If a custom relevance_func is provided it must have this signature:
+        `Callable[[np.ndarray, np.ndarray], np.ndarray]`
+        It should accept X, y as arguments and it should compute the score for each feature of X
+        and return an array of shape (n_features_in_,).
+    !! warning:
+        If a custom redundancy_func is provided it must have the same signature as the method _redundancy_pearson
+
+    Attributes
+    ----------
+    _y_dtype : np.dtype
+        data type of y
+    selected_features_ : array-like of shape (k,)
+        Indexes of the selected features.
+    scores_ : array-like of shape (k,)
+        Scores of the selected features.
+
+    Examples
+    --------
+    ```py
+    from sklego.feature_selection.mrmr import MaximumRelevanceMinimumRedundancy
```

Comment:
Can import directly from `feature_selection` module:

```suggestion
    from sklego.feature_selection import MaximumRelevanceMinimumRedundancy
```